data/jakarta/test-security-services-proxy-certs.sh

```diff
@@ -108,14 +108,15 @@ snap_wait_port_status 50025 open
 # We are running the test script with 'sudo' and although the edgeca snap has the home interface, 
 # which allows access to the home directory, when running as sudo, the user is root, 
 # so it has a different home directory and doesn't have write access to your home directory. 
-# The simplest fix: use `su - "$USER" -c "command"`
-su - "$USER" -c "edgeca gencsr --cn localhost --csr csrfile --key csrkeyfile"
-su - "$USER" -c "edgeca gencert -o localhost.cert -i csrfile -k localhost.key"
-snap set edgexfoundry env.security-proxy.tls-certificate="$(cat localhost.cert)"
-snap set edgexfoundry env.security-proxy.tls-private-key="$(cat localhost.key)"
-
-# verify CA-signed TLS certificate
-code=$(curl --insecure --silent --include \
+# It's therefore easiest to use the $SNAP_DATA directory of the EdgeCA snap:
+EDGECA_DIR="/var/snap/edgeca/current/"
+edgeca gencsr --cn localhost --csr $EDGECA_DIR/csrfile --key $EDGECA_DIR/csrkeyfile
+edgeca gencert -o $EDGECA_DIR/localhost.cert -i $EDGECA_DIR/csrfile -k $EDGECA_DIR/localhost.key
+snap set edgexfoundry env.security-proxy.tls-certificate="$(cat $EDGECA_DIR/localhost.cert)"
+snap set edgexfoundry env.security-proxy.tls-private-key="$(cat $EDGECA_DIR/localhost.key)"
```

Comment:
These `snap set` commands don't respect the exit code of `cat` and instead perform certificate and private-key removal operations:
```
$ x="$(cat localhost.cert)"
cat: localhost.cert: No such file or directory
$ echo $?
1

# compare to:
$ sudo snap set edgexfoundry env.security-proxy.tls-certificate="$(cat localhost.cert)"
cat: localhost.cert: No such file or directory
$ echo $?
0
```
This is probably a bug from `snap set`.

I can think of two ways to check:
1. Read the files and output to variables, then read from variable
2. Check if the files exists before directly reading from them and setting the config variables

Reply:
Could these be replaced by:
```
JWT_FILE=/var/snap/edgexfoundry/current/secrets/security-proxy-setup/kong-admin-jwt
JWT=`sudo cat ${JWT_FILE}`
edgexfoundry.secrets-config proxy adduser --token-type jwt --user user01 --algorithm ES256 --public_key public.pem --id USER_ID --jwt ${JWT} 
```
?

Reply:
I updated this so that it checks if the files exist, then reads them to variables and uses those.